### DISCUSSION TITLE
Why are permissions explicitly granted to anon and autenticated roles for objects created by supabase_admin

### DISCUSSION THREAD

**Discussion by chipilov** [supabase/supabase#4834] [Questions] [answered]
I noticed that the PostgreSQL instances supplied by Supabase explicitly grant permissions to ```anon``` and ```authenticated``` roles for objects created by ```supabase_admin```: 

https://github.com/supabase/supabase/blob/68aca6cd61f1471947eb34de4fd3ac66ce37f394/docker/volumes/db/init/00-initial-schema.sql#L38

What is the reason for this? Is it because Supabase users tend to use the Supabase SQL editor from the dashboard to create objects and that editor uses supabase_admin? 

I ask because I would like to revoke such permissions and I am wondering if anything can break if I do that? NOTE, that in my workflow I DO NOT create objects with the ```supabase_admin``` role, so that should NOT be a problem - I am wondering if there are any other scenarios I should be aware of.

**Comment by awalias** [comment, marked as answer]
yes I believe this is the primary reason, if we didn't grant this permission then any tables created via the dashboard would not be accessible via the API (posgrest)

**Comment by lwolle** [comment]
Just asking, wouldn't it make sense to run the supabase cli and studio using the same postgres user/role. I've started using supabase a few weeks ago and have seen this and related questions popping up here and there several times now. Is it already part of the documentation?

**Reply by GaryAustin1** [reply]
The user should now be Postgres for anything you do in studio/cli/sql editor.  The API will still use other roles as you want them secure from doing things they should not.  By default tables/functions/etc created in the public schema will be granted to those roles also.
